ee/webhooks/internal/components/webhook_controller/controllers/utils/utils.go

```diff
@@ -0,0 +1,180 @@
+package utils
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type ErrorType string
+
+const (
+	NoneType       ErrorType = "NONE"
+	ValidationType ErrorType = "VALIDATION_TYPE"
+	NotFoundType   ErrorType = "NOT_FOUND"
+	InternalType   ErrorType = "INTERNAL_TYPE"
+)
+
+type Response[T interface{}] struct {
+	T    ErrorType
+	Err  error
+	Data *T
+}
+
+func SuccessResp[T interface{}](d T) Response[T] {
+	return Response[T]{
+		T:    NoneType,
+		Err:  nil,
+		Data: &d,
+	}
+}
+
+func ValidationErrorResp[T interface{}](err error) Response[T] {
+	return Response[T]{
+		T:    ValidationType,
+		Err:  err,
+		Data: nil,
+	}
+}
+
+func InternalErrorResp[T interface{}](err error) Response[T] {
+	return Response[T]{
+		T:    InternalType,
+		Err:  err,
+		Data: nil,
+	}
+}
+
+func NotFoundErrorResp[T interface{}](err error) Response[T] {
+	return Response[T]{
+		T:    NotFoundType,
+		Err:  err,
+		Data: nil,
+	}
+}
+
+const (
+	ErrValidation  = "VALIDATION_TYPE"
+	ErrHealthcheck = "HEALTHCHECK_STATUS"
+)
+
+func DecodeJSONBody(r *http.Request, v any) error {
+	data, err := io.ReadAll(r.Body)
```

Comment:
Quick opti, you can use json.NewDecoder(r.Body).Decode, instead of io.readall and unmarshall. It has its own buffering and for big body, it's actually better in performances than to raed everything at once in memory

Reply:
Done